pylint/checkers/variables.py

```diff
@@ -585,7 +585,12 @@ def get_next_to_consume(self, node: nodes.Name) -> list[nodes.NodeNG] | None:
             and parent_node.iter == node
             and parent_node.target in found_nodes
```

Comment:
```suggestion
            and parent_node.target in found_nodes
            and any(...)
```
This would be more in line with the style in the functional and permits to not allocate other_definitions and also shortcut at the first elements.

Reply:
Hi @Pierre-Sassoulas, do you mean the following change?
```python
if (
    found_nodes
    and isinstance(parent_node, nodes.For)
    and parent_node.iter == node
    and parent_node.target in found_nodes
    and any(fn != parent_node.target for fn in found_nodes)
):
    found_nodes = [fn for fn in found_nodes if fn != parent_node.target]
elif (
    found_nodes
    and isinstance(parent_node, nodes.For)
    and parent_node.iter == node
    and parent_node.target in found_nodes
):
    found_nodes = None
```
It will make the code a bit longer and we will require two iterations over `parent_node` but I can change it to this.

Reply:
More like this : https://github.com/pylint-dev/pylint/pull/10569/files#r2360994187